### PR TITLE
[Gutenberg] Audio block capability now enables/disables media upload's media sources

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes  https://github.com/wordpress-mobile/gutenberg-mobile/pull/3523

`WordPress Android` wordpress-mobile/WordPress-Android#14680
`Gutenberg Mobile` wordpress-mobile/gutenberg-mobile#3523
`Gutenberg` https://github.com/WordPress/gutenberg/pull/31966

## Description

This PR checks to see if the user's site is on a free, paid or self hosted plan and as such, the relevant media upload options are shown. 

| Free Plans      	| Paid Plans              	| Self Hosted             	|
|-----------------	|-------------------------	|-------------------------	|
| Insert from URL 	| Insert from URL         	| Insert from URL         	|
|        -        	| Choose from device      	| Choose from device      	|
|        -        	| WordPress Media Library 	| WordPress Media Library 	|

### WP.com sites

**Note:** 

Support for Audio block audio uploads, and access to the media library is restricted to sites that are on a paid plan.

#### Free Plan
1. Create a new post. 
2. Add the Audio block. 
3. Notice that only the `Insert from URL` option is available as users aren't able to upload audio on free plans. 

#### Paid plan
1. Create a new post. 
2. Add the Audio block. 
3. Notice that all media options are available as users are able to upload audio on paid plans. 

Free Plan | Paid Plan 
--------|-------
 <kbd></kbd> |   <kbd></kbd>

### Self hosted sites

1. Create a new post. 
2. Add the Audio block. 
3. Notice that all media options are available as users are able to upload audio on self hosted sites. 

<kbd></kbd>

## Regression Notes
1. Potential unintended areas of impact

The blocks Cover, File, Gallery, Video, Media & Text all utilise the `media-upload` component that was modified. Verify that these blocks function as expected and all the relevant options are still shown for them in both paid and free plans. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
There are existing tests that are doing uploads via the WP media library and the file system so if there was an issue with the implementation those tests would fail. 

3. What automated tests I added (or what prevented me from doing so)
N/A

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ extension GutenbergViewController: GutenbergBridgeDataSource {
             .contactInfoBlock: post.blog.supports(.contactInfo) && FeatureFlag.contactInfo.enabled,
             .unsupportedBlockEditor: isUnsupportedBlockEditorEnabled,
             .canEnableUnsupportedBlockEditor: post.blog.jetpack?.isConnected ?? false,
-            .audioBlock: !isFreeWPCom, // Disable audio block until it's usable on free sites via "Insert from URL" capability
+            .audioBlock: !isFreeWPCom,
```
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: Related to this [comment](https://github.com/WordPress/gutenberg/pull/31966#discussion_r635667170), what do you think about renaming the capability?

**Comment by jd-alexander** [comment]
This PR will be updated once https://github.com/wordpress-mobile/gutenberg-mobile/pull/3548 is merged in so a final `Gutenberg Mobile PR` wordpress-mobile/gutenberg-mobile#3523 can be generated for testing on both platforms.

**Comment by fluiddot** [comment]
> This PR will be updated once [wordpress-mobile/gutenberg-mobile#3548](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3548) is merged in so a final `Gutenberg Mobile PR` [wordpress-mobile/gutenberg-mobile#3523](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3523) can be generated for testing on both platforms.

@jd-alexander Heads up that the [PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3548) is already merged 🎊 .

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift (hunk 1 above)
Comment (jd-alexander) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: Done here 2d20967

**Review by fluiddot — COMMENTED** [review]
I tested on iPhone 11 (iOS 14.2) and works nice 🎊 !

Once the Gutenberg PR is approved I'll do a second run and approve it.

**Comment by fluiddot** [comment]
> I tested on iPhone 11 (iOS 14.2) and works nice 🎊 !
> 
> Once the Gutenberg PR is approved I'll do a second run and approve it.

After further testing, I've spotted an error when uploading an audio file via the "Other Apps" option, [here is the comment on GB-mobile with all the details](https://github.com/WordPress/gutenberg/pull/31966#issuecomment-849693150).

**Comment by jkmassel** [comment]
We're cutting `17.5` today, so this PR is being bumped to `17.6`

**Comment by fluiddot** [comment]
Looks like CircleCI didn't post the comment referencing the installable builds so here it goes.

**Latest installable build:** `49354`.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

Approved via https://github.com/WordPress/gutenberg/pull/31966#pullrequestreview-673060329.

**Pull request merged**